llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -259,3 +264,14 @@ bool AIE2PassConfig::addInstSelector() {
     addPass(createMachineFunctionDumperPass(/*Suffix=*/"after-isel"));
   return false;
 }
+
+unsigned
+AIE2TargetMachine::getAddressSpaceForPseudoSourceKind(unsigned Kind) const {
+  switch (Kind) {
+  case PseudoSourceValue::Stack:
+  case PseudoSourceValue::FixedStack:
+    return StackAddrSpace;
+  default:
+    return static_cast<unsigned>(AIE2::AddressSpaces::none);
```

Comment:
Based on my comment https://github.com/Xilinx/llvm-aie/pull/95#discussion_r1713409406 I think we should return `AIE2::AddressSpaces::all` in the default case, and handle `TileMemory` Kind here as well.

Reply:
Here we need to return the default address space that is 0 which is AIE2::AddressSpaces::none.
This will translate into reserving all the bankbits.